### ISSUE TITLE
feature request: reporting results in rmarkdown files using subtitle helper functions

### ISSUE THREAD

**Issue by IndrajeetPatil** [IndrajeetPatil/statsExpressions#1]
> I would like to know if there's a way to use the APA-formatted text of the statistical results prepared with the subtitle helper functions in a plain RMarkdown file (probably with inline R code?)

For example-
```rmarkdown
The correlation analysis showed that `x` and `y` were correlated:
 `r subtitle_ggscatterstats(data, x, y)`.
```

Also see:
https://easystats.github.io/report/

**Comment by matcasti** [comment]
Yes please, this would be awesome for report writing. I've check the code inside some of your packages and it seems that you could pull it out smoothly.
Best regards
Matías

**Comment by matcasti** [comment]
Working from the shoulders of the previous example code, this function maybe can accomplish the task related to #104 

```r
format_markdown <- function(expr) {

  # trasnform expression to be able to modify it
  expr <- as.list(x = as.list(expr)[[1]])

  # replace invalid patterns to be evaluated
  expr <- gsub(")[", ") * \"\"[", expr, fixed = TRUE)
  expr <- gsub("](", "] * list2(", expr, fixed = TRUE)

  # transform to again to expression to evaluate it
  expr <- lapply(expr, str2lang)
  expr <- as.call(expr)

  # global variables; could be changed for anything else
  p <- "p"; CI <- "CI"; chi <- "*X*";
  mu <- "*mu*"; log <- "log"; BF <- "BF";
  e <- "e"; epsilon <- "Epsilon"; R <- "R";
  HDI <- "HDI"; xi <- "xi"; omega <- "Omega";

  # list works pasting expressions
  list <- function(...) paste(..., sep = ", ")

  # `list2` works the same but only for expressions with >= 1 parameter(s)
  list2 <- function(...) paste0("(", paste(..., sep = ", "), ")")

  # italic text just has stars around it
  italic <- function(s) paste0("*", s, "*")

  # wide hat has no effect on final output
  widehat <- function(s) as.character(s)

  # single subscripts are entered using subsetting
  `[` <- function(main, subscript) paste0(main, "~", subscript, "~")

  # single superscript are entered using symbol in both sides
  `^` <- function(main, superscript) paste0(main, "^", superscript, "^")

  # this symbol will concatenate
  `*` <- function(lhs, rhs) paste0(lhs, rhs, collapse = ", ")

  # replace to equal sign
  `==` <- function(lhs, rhs) paste0(lhs, " = ", rhs)

  # suppress formula behaviour
  `~` <- function(lhs, rhs) paste0(lhs, rhs)

  # evaluate the expression to produce a string
  eval(expr)
}
```

With this, a got the next outputs:

## gghistostats

```r
format_markdown(
  ggstatsplot::gghistostats(
  data = iris,
  x = Sepal.Length,
  test.value = 7,
  output = "subtitle"
  )
)
#> [1] "*t*~Student~(149) = -17.11, *p* = 5.56e-37, *g*~Hedges~ = -1.39, CI~95%~[-1.62, -1.17], *n*~obs~ = 150"
```

## ggdotplotstats

```r
format_markdown(
  ggstatsplot::ggdotplotstats(
    data = iris,
    y = Species,
    x = Sepal.Length,
    type = "np",
    output = "subtitle"
  )
)
#> [1] "*V*~Wilcoxon~ = 6.00, *p* = 0.25, *r*~biserial~^rank^ = 1.00, CI~95%~[1.00, 1.00], *n*~obs~ = 3"
```

## ggbetweenstats

```r
format_markdown(
  ggstatsplot::ggbetweenstats(
    data = iris,
    x = Species,
    y = Sepal.Length,
    type = "b",
    output = "subtitle"
  )
)
#> [1] "log~e~BF~01~ = -65.10, *R^2^*~Bayesian~^posterior^ = 0.61, CI~95%~^HDI^[0.54, 0.67], *r*~Cauchy~^JZS^ = 0.71"
```
```r
format_markdown(
  ggstatsplot::ggbetweenstats(
    data = iris,
    x = Species,
    y = Sepal.Length,
    type = "r",
    output = "subtitle"
  )
)
#> [1] "*F*~trimmed-means~(2, 53.84) = 111.95, *p* = 0.00, xi = 0.85, CI~95%~[0.77, 0.91], *n*~obs~ = 150"
```
```r
format_markdown(
  ggstatsplot::ggbetweenstats(
    data = iris,
    x = Species,
    y = Sepal.Length,
    type = "p",
    output = "subtitle"
  )
)
#> [1] "*F*~Welch~(2, 92.21) = 138.91, *p* = 1.51e-28, Omega~p~^2^ = 0.74, CI~95%~[0.67, 1.00], *n*~obs~ = 150"
```

**Comment by roaldarbol** [comment]
First of all, thanks for developing these amazing packages!

This feature would still be immensely helpful to get a solution to! Most results section in journal papers requires results to be written within the text, and this could provide a perfect way to have reproducible results section, as the stats would update in case the underlying data does. IMO this should be a quite high priority. :-) 
Most Markdown formats understand TeX `$` math syntax these days, so I think that's probably the best way to go. I see two potential ways to go about this:
1. Make a plotmath-to-TeX converter which outputs an expression inside `$`s.
2. Re-factor the code to generate TeX by default, and then convert to plotmath for the graphs (there exists a converter for that: [latex2exp](https://www.stefanom.io/latex2exp/)).

As an aside, also note the new [`{marquee}`](https://marquee.r-lib.org/) which enables seamless Markdown rendering within ggplots. They do not yet support LaTeX equations, but Thomas seems to be on it (https://github.com/r-lib/marquee/issues/2). So once that works, outputting the TeX expressions in  and rendering them in plots with {marquee} might be the least work in the long run and probably the most extensible way.

**Comment by IndrajeetPatil** [comment]
@roaldarbol It's uncanny that you wrote this today, since it's only yesterday I subscribed to the `{marquee}` issue you mention here 👀 

Thanks for the detailed heads-up, though. This is definitely something I am going to follow closely, and hopefully it becomes easier to support LaTeX equations in marquee, which would also make it easier to support them in `{statsExpressions}`.

**Comment by roaldarbol** [comment]
Haha, uncanny timing indeed! Fingers crossed they manage to enable it soon! :-D

**Comment by IndrajeetPatil** [comment]
Thank you to @matcasti and everyone who has contributed ideas and code here — this feature request has clearly resonated with the community.

After reflection, I've decided to close this PR and the associated issue. While the ability to convert statistical expressions to Markdown/LaTeX for inline reporting is genuinely useful, I think it's better suited for a **standalone extension package** (e.g., `{statsExpressionsExtra}` or similar) rather than being folded into the core `{statsExpressions}`.

I should also be transparent: I no longer work in R professionally, which means I need to be disciplined about keeping the maintenance demands of this package as low as possible. Introducing significant new functionality — however valuable — doesn't align with that goal at this stage.

The good news is that building such an extension package has never been easier. Coding agents (like GitHub Copilot, Claude, etc.) can scaffold a full R package in minutes, so if anyone in the community is motivated to pursue this, the barrier is now very low. The proof-of-concept code in this PR and in the issue comments is an excellent starting point.

Thanks again for your patience and enthusiasm.
